flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClient.java

```diff
@@ -205,7 +205,7 @@ public static void submitJobDetached(
 		Preconditions.checkNotNull(timeout, "The timeout must not be null.");
 
 		Future<Object> future = jobManagerGateway.ask(
-				new JobManagerMessages.SubmitJob(jobGraph, false),
+				new JobManagerMessages.SubmitJob(jobGraph, false, jobGraph.getClientVersion()),
```

Comment:
Why don't you simply use the `getImplementationVersion` of the `JobClient` but the version of the `Configuration` which is part of the `jobGraph` instance?